global/autoload/audio/audio_queue.gd

```diff
@@ -0,0 +1,49 @@
+class_name AudioQueue extends Node
+
+@export var stream_player_count: int = 4
+@export_enum(&"Master", &"Music", &"SFX") var bus: String = &"Master"
+
+var available_stream_players: Array[AudioStreamPlayer] = []
+var audio_queue: Array[Dictionary] = []
+
+###############
+## overrides ##
+###############
+
+
+func _ready() -> void:
+	for i in stream_player_count:
+		var stream_player: AudioStreamPlayer = AudioStreamPlayer.new()
+		add_child(stream_player)
+		stream_player.finished.connect(_on_stream_player_finished.bind(stream_player))
+		available_stream_players.append(stream_player)
+
+
+func _process(_delta: float) -> void:
```

Comment:
I like the queue, but not sure if `_process` is necessary? can we use "audio finished" signals? 🤔 

(I have a particle queue `res://scenes/component/effect/label_effect_queue/label_effect_queue.gd` where I do pop_front() on timer signal.)

Reply:
It looks like that code is for processing a task every second, whereas we have 8 (this number is customizable) audio players at once and they pick up sounds from the queue to play as they are available.

Also, I think using a timer is computationally more expensive than using _process.

I may be able to get it working without  _process or a timer by using signals, do you think it's worth exploring?

Reply:
In this case, I was thinking to use `finished` signal of `AudioStreamPlayer` 

https://docs.godotengine.org/en/stable/classes/class_audiostreamplayer.html#signals

A - When a stream player finishes playing the sound, it picks a new one from the `sound` queue if it is not empty.

B - When `play` is called, assign the sound to an available player or if none are, add it to `sound` queue to be picked up as soon as one finishes and becomes available.

No need for timers or _process.

You can connect `finished` signal of each `.new()` stream player to a generic handler method to handle A.

What do you think?

Reply:
Yeah that's what I was going to try, I'll swap over to that